homeassistant/components/modbus/climate.py

```diff
@@ -209,7 +209,7 @@ def target_temperature_step(self):
     def set_temperature(self, **kwargs):
         """Set new target temperature."""
         target_temperature = int(
-            (kwargs.get(CONF_TEMPERATURE) - self._offset) / self._scale
+            (kwargs.get(ATTR_TEMPERATURE) - self._offset) / self._scale
```

Comment:
Side note: This will error if the dict.get return None.

Reply:
I am aware of that, we have a number of these cases in the code. I have a PR in progress where I try to chase them all.

thanks.